### ISSUE TITLE
mahdikarimi - Attacker can DoS redeeming of PT and YT tokens

### ISSUE THREAD

**Issue by sherlock-admin** [sherlock-audit/2024-01-napier-judging#73]
mahdikarimi

high

# Attacker can DoS redeeming of PT and YT tokens

## Summary
Attacker can force stake bufferETH to DoS of adapter and redeeming of tranche shares temporarily and repeat the process infinitely to create a permanent DoS and lock of funds. 
## Vulnerability Detail
in baseLSTAdapter there is a variable called befferETH which is amount ETH available in adapter ( not staked ) to facilitate withdraw and redeeming of shares from vault, also there is a targetBufferPercentage which is the percentage of total assets that shouldn't be staked so the vault has enough funds to pay shareholders in case of redeeming and withdraw, if withdraw amount is more than bufferETH transaction would revert  and user needs to wait for recovering bufferETH, until withdrawal request is claimed from lido or frax , also if amount of bufferETH is lower than targetBufferPercentage some or all of deposit amounts would have been added to bufferETH to reach targerBuffer.
however a user can perform a prefundedDeposit and preFundedRedeem in one transaction to force stake bufferETH and perform it repeatedly to DoS all tranches that integrate with it.
there are two scenarios for this  
1 - bufferETH is equal or more than targerBuffer : Consider bufferETH value is 200 ETH, user deposits 200 ETH and receive 200 shares, the deposited amount would be staked into lido or frax since bufferETH is enough, user withdraws shares immediately, and since there is no fee charged on deposit and withdraw from the adapter, he would be paid 200 ETH from bufferETH, now bufferETH is 0 which means no asset for other users to withdraw.
2 - bufferETH is lower than targerBuffer : Consider bufferETH value is 200 ETH, and targetBufferETH is 400, user deposits 400 ETH and receives 400 shares, 200 ETH would be added to bufferETH and 200 would be staked now bufferETH is 400, user redeems shares immediately, and since there is no fee charged on deposit and withdraw in the adapter, he would be paid 400 ETH from bufferETH, now bufferETH is 0 which means no asset for users to withdraw.
## Impact
Since multiple tranches use adapter, this can temporarily DoS all integrated tranches, attack can be performed manually, cost of attack is low, attacker needs to pay only gas fee, and a small amount due to round down of shares however it's very small probably some wei so we can consider only gas fee, a bot can perform this everytime that there is balance in adapter to permanently DoS of protocol, in this case attacker should be front-run to freeing of funds but we can't expect normal users front-run attacker, it seems there is no profit for attacker, it would be easier to DoS after tranche maturity because there is no mint so attacker needs to perform it only after claim withdraw request from lido or frax, also it would melt the assets since every withdraw request from lido or frax would decrease asset amount ( since ETH/stETH is not 1:1) .  
## Code Snippet
https://github.com/sherlock-audit/2024-01-napier/blob/main/napier-v1/src/adapters/BaseLSTAdapter.sol#L93-L136
https://github.com/sherlock-audit/2024-01-napier/blob/main/napier-v1/src/adapters/BaseLSTAdapter.sol#L156
https://github.com/sherlock-audit/2024-01-napier/blob/main/napier-v1/src/adapters/lido/StEtherAdapter.sol#L59
## Tool used

Manual Review

## Recommendation
add enable/disable stake functionality in the adapter so that rebalancer or admin can disable staking of tokens which prevent lock of funds.

Duplicate of #120

**Comment by MehdiKarimi81** [comment]
Escalate 
This is a duplicate of #89.

**Comment by sherlock-admin2** [comment]
> Escalate 
> This is a duplicate of #89.

You've created a valid escalation!

To remove the escalation from consideration: Delete your comment.

You may delete or edit your escalation comment anytime before the 48-hour escalation window closes. After that, the escalation becomes final.

**Comment by nevillehuang** [comment]
Agree, this is a duplicate of #89

**Comment by Czar102** [comment]
@cvetanovv would also like to get to know your stance before accepting the escalation.

**Comment by cvetanovv** [comment]
I agree with the escalation. This is a duplicate of #89.

**Comment by Czar102** [comment]
Planning to accept the escalation and make this a duplicate of #89.

**Comment by Czar102** [comment]
Result:
High
Duplicate of #89

Please note that the severity may be further altered depending on the outcome of the escalation on #89.

**Comment by sherlock-admin4** [comment]
Escalations have been resolved successfully!

Escalation status:
- [MehdiKarimi81](https://github.com/sherlock-audit/2024-01-napier-judging/issues/73/#issuecomment-1984329078): accepted

**Comment by sherlock-admin4** [comment]
The protocol team fixed this issue in PR/commit https://github.com/napierfi/napier-v1/pull/180.

**Comment by sherlock-admin4** [comment]
The Lead Senior Watson signed off on the fix.
